chef_master/source/resource_apt_package.rst

```diff
@@ -302,11 +302,30 @@ The chef-client will determine the correct provider based on configuration data
 Generally, it's best to let the chef-client choose the provider, and this is (by far) the most common approach. However, in some cases, specifying a provider may be desirable. There are two approaches:
 
 * Use a more specific short name---``yum_package "foo" do`` instead of ``package "foo" do``, ``script "foo" do`` instead of ``bash "foo" do``, and so on---when available
-* Use the ``provider`` property within the resource block to specify the long name of the provider as a property of a resource. For example: ``provider Chef::Provider::Long::Name``
+* Use ``build_resource``. This replaces all previous use cases where the provider class was passed in through the ``provider`` property:
+
+  .. code-block:: ruby
+
+     pkg_resource = case node['platform_family']
+       when "debian"
+         :dpkg_package
+       when "fedora", "rhel", "amazon"
+         :rpm_package
+       end
+
+     pkg_path = ( pkg_resource == :dpkg_package ) ? "/tmp/foo.deb" : "/tmp/foo.rpm"
+
+     build_resource(pkg_resource, pkg_path) do
```

Comment:
should be `declare_resource`.

`build_resource` doesn't put it into the resource collection.  you'd use that if you wanted to build a resource, then later explicitly call an action on with `run_action(:whatever)` which should not be a common use case worth documenting.  generally you want to stick to encouraging people to always call `declare_resource` (in fact `build_resource` may be one of those APIs that really shouldn't be put in front of customers since it'll confuse them and be wrong in 99% of their cases).